server/api/itemRouter.js

```diff
@@ -0,0 +1,115 @@
+const Item = require('../db/models/item');
+const Category = require('../db/models/category');
+const express = require('express');
+const router = express.Router();
+
+/**
+ * Add an item
+ * 
+ * req.body: {sourceLink: String,
+ *            highlight: String,
+ *            userId: mongoose.ObjectId,
+ *            placesId: id(optional),
+ *            comment: String (optional),}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.post('/addItem', (req, res, next) => {
+  const newItem = req.body;
+  const item = new Item(newItem);
+  let response = {}; 
+  item.save((err) => {
+    if (err) {
+      response.success = false;
+      response.message = 'error adding item';
+      res.json(response);
+      next(err);
+    } else {
+      response.success = true;
+      response.message = 'item added';
+      res.json(response);
+    }
+  });
+});
+
+/**
+ * Get all items
+ * 
+ * req.body: {userId: mongoose.objectId}
+ * 
+ * response: [{sourceLink: String,
+ *            highlight: String,
+ *            placeId: id(optional),
+ *            comments: String (optional),}]
+ */
+router.get('/getItems', (req, res, next) => {
+  Item.find({ userId: req.body.userId }, (err, items) => {
+    if (err) {
+      next(err);
+    }
+    if (items) {
+      res.json(items);
+    }
+  });
+});
+
+/**
+ * Add item to category
+ * 
+ * req.body: {itemId: mongoose.objectId,
+ *            categoryId: mongoose.objectId}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.put('/addItemToCategory', async (req, res, next) => {
+  let response = {}; 
+  try {
+    const itemId = req.body.itemId;
+    const categoryId = req.body.categoryId;
+    await Item.update({ _id: itemId }, { $push: { categoryIds: this.categoryId } }, done).exec();
+    await Category.update({ _id: categoryId }, { $push: { items: this.itemId } }, done).exec();
```

Comment:
Would it be worth adding one more layer of verification by comparing the user ID of the item and category to the one in the request?

Reply:
I agree this is safe, but verification seems to fit in a top level common middleware, applicable for a few handlers. Let's put a TODO and open an issue on this.

Reply:
@ricazhang Oh.. I see what you mean. 

Is this check necessary considering all itemIds and categoryIds should be completely unique, so it would be unlikely that we get the wrong item or category belonging to a different user unless the request coming from the client-side is wrong, which could be possible but just our own mistake? Or is it better practice to do this check? 

I added TODO and opened an issue on this @Alfredhgoog